vispy/visuals/axis.py

```diff
@@ -392,9 +458,133 @@ def _get_tick_frac_labels(self):
             length = self.axis.pos[1] - self.axis.pos[0]  # in logical coords
             n_inches = np.sqrt(np.sum(length ** 2)) / transforms.dpi
 
-            major = _get_ticks_talbot(domain[0], domain[1], n_inches, 2)
+            major = _get_ticks_talbot(domain[0], domain[1], n_inches, density)
+
+            labels = []
+
+            if self.axis.axis_mapping is not None:
+                for x in major:
+                    x = round(x, 10)
+                    if x >= 0:
+                        try:
+                            label = self.axis.axis_mapping[int(x)]
+                            label_plus_one = self.axis.axis_mapping[int(x) + 1]
+                        except IndexError:
+                            label = x
+                            label_plus_one = None
+                            print("Axis mapping ran out of data at index %s use update_axis_mapping() in AxisVisual "
+                                  "to extend the mapping range in Runtime or map a longer list/array in the first "
+                                  "place" % str(int(x)))
```

Comment:
It looks this should maybe be a `warnings.warn`? When does this condition get encountered? Will this be a common occurrence if people are panning/zooming on a plot?

Reply:
Good question I'm also not quite sure. zooming out can obviously cause this.
I was thinking about interpolating the dates once the mapping runs out of entries. i.e. self.axis.axis_mapping[last] - self.axis.axis_mapping[2nd_to_last]
then for x = last+ n -> self.axis.axis_mapping[last] + n * (self.axis.axis_mapping[last] - self.axis.axis_mapping[2nd_to_last])

This would only work if the mapping has a constant delta!

In case of people not mapping dates, something like strings I have no clue what to do - probebly a warnings.warn would be good